docs/rfc/rfc-017-abci++-vote-extension-propag.md

```diff
@@ -553,6 +554,32 @@ Then, we need to handle two new situations, roughly equivalent to cases (h.1) an
   However we can manually have the node crash and recover as a workaround. This effectively converts
   this case into (h.1).
 
+### Upgrade Path
+
+v0.36 of Tendermint will be the first version to implement the vote extension system.
+The v0.36 release aims to be backwards compatible with the previous release, v0.35,
+which had no implementation of vote extensions and no mechanisms in place to facilitate
+their later addition.
+
+Vote extensions pose an issue for Tendermint upgrades. Chains that upgrade from
+v0.35 to v0.36 will attempt to produce the first height running v0.36 without vote
+extension data from the previous height. We intend to allow chains to _require_
+vote extensions data. Chains that do so will not make progress without vote
```

Comment:
Can you please add a reference to Solution 3, stating something like: the key property of Solution 3 above must be broken in order to upgrade.

Reply:
Added, please take a look.